### DISCUSSION TITLE
groupBy returns internal server error for nested relations

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#13999] [Feature Requests]
### Describe the Bug

I tried to use groupBy to filter on belong_user.id, but http returned 500

### To Reproduce

```typescript
export class Nearby {
    static async getUserList() {
        const params = {
            fields: ["*", "belong_user.id"],
            sort: ["-date_created"],
            // filter: {
            //     belong_user: {
            //         _neq: "$CURRENT_USER"
            //     }
            // },
            groupBy: ["belong_user.id"]
        }

        const data = await Network.get("items/location", params)

        console.log((await data.response.json()).data)
    }
}
```

<img width="948" alt="image" src="https://user-images.githubusercontent.com/101538735/174431820-a971228c-0b97-46d3-9521-943195bf1fe4.png">

If I cancel groupBy, the return result will be duplicated, I want the result to be belong_user only return a location record

<img width="996" alt="image" src="https://user-images.githubusercontent.com/101538735/174431910-8851f704-31d4-429f-ae69-725d07ca0136.png">

### Errors Shown

An unexpected error occurred.

### What version of Directus are you using?

9.12.2

### What version of Node.js are you using?

18

### What database are you using?

Postgres 14

### What browser are you using?

Chrome

### How are you deploying Directus?

locally

**Comment by br41nslug** [comment]
As a temporary workaround grouping by the top level field does work for me `groupBy: ["belong_user"]`

**Comment by rijkvanzanten** [comment]
Grouping on nested m2o values isn't currently supported. Converting this to a feature request 👍🏻

**Reply by mojerro** [reply]
Is there any plan to support this feature~

**Reply by mojerro** [reply]
I think it will be a very convenient feature for getting aggregated data

**Comment by Eliophot** [comment]
You can use lodash : https://lodash.com/docs/4.17.15#groupBy
```
_.groupBy(response.data, 'languages_id');
```

**Reply by mojerro** [reply]
It's quite slow if there is a large sum of data because you need to fetch all data to the browser
